### PR TITLE
[ BB2-1472 ] Update readme.md

### PR DESCRIPTION
**Pull request by gmaciolek**
**JIRA Ticket:**
[BB2-1472](https://jira.cms.gov/browse/BB2-1472)

**User Story or Bug Summary:**

Polish the GitHub SDK repo and readme docs related to Node & Python SDKs.  Proof-read and check styling.

### What Does This PR Do?

Revise readme to improve organization of information, make instructions more concise and clear, add sections commonly included in readme files (Prerequisites, About, Help and Support), and apply DASG content style guide-based changes.

### What Should Reviewers Watch For?

If you're reviewing this PR, please check these things, in particular:

- Is the information accurate?
- Is the information clearly communicated?
- Is any critical information missing?
- Is any trivial information or information that might be better addressed in a separate documented included?

Questions for reviewers:
- Can we link the FHIR version formats somewhere for more detail? Or do we assume readers know what the difference is between the 2 formats?
- If we want to encourage people to use the V2 data, do we want to de-emphasize V1? For example, we could only present V2 information, but include a note that V1 is also available.

Final note about a note:
In the Usage section, there's a note at the top because Adrian is going to provide some intro text to give more context and detail about what's included in the code snippet. I pointed out that it's hard to parse out the developer notes within the code block to understand what this code is and does so maybe it could be broken out into smaller chunks of code. Adrian noted that it's a good experience to be able to grab the whole chunk of code because you can paste once and run with it. To meet in the middle, we decided some intro text would be helpful.

### What Security Implications Does This PR Have?
none

### I have gone through and verified that...:
**I'm not sure that this checklist is directly applicable to a documentation-specific type of PR, but I'll check off what seems relevant. Let me know if there's anything else I should verify.**

- [ ] This PR is reasonably limited in scope, to help ensure that:
  1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
  2. There isn't too much of a burden on reviewers.
  3. Any problems it causes have a small "blast radius".
  4. It'll be easier to rollback if that becomes necessary.
- [x] I have named this PR and its branch such that they'll be automatically be linked to the (most) relevant Jira issue, per: <https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html>.
- [x] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
- [ ] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
- [ ] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
- [x] Reviews are requested from both:
  - At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
  - Any relevant engineers on other projects (e.g. BFD, SLS, etc.).
- [ ] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
  - Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

**Comment by ajshred** [comment]
Yea Gretchen!  You did it!!!

**Review by ajshred — APPROVED** [review]
Hi Gretchen!  This looks great!  I added a short intro where you asked and pushed that up for you to "tweak" if necessary.  Thank you!

File: readme.md [hunk 1]
```diff
@@ -1,111 +1,110 @@
...
 
-the configuration is in json format and stored in a local file, the default location
-is current working directory with file name: .bluebutton-config.json
+- Sandbox location: https://sandbox.bluebutton.cms.gov
+- Production location: https://api.bluebutton.cms.gov
 
-A sample configuration json:
+Version data formats:
+
+- V1: FHIR STU2
```
Comment (JFU-GIT) [inline readme.md:70]: STU3

**Review by dtisza1 — APPROVED** [review]
@gmaciolek This looks good to me!

I like the improvements! I especially like the example code intro addition.

Good work on this!

**Review by JFU-GIT — APPROVED** [review]
Looks good!

File: readme.md [hunk 2]
```diff
@@ -1,111 +1,109 @@
...
+}
 
-e.g. for backoff_factor is 5 seconds, it will generate wait intervals: 2.5, 5, 10, ...
+```
 
-to disable the retry: set total = 0
+## Usage <a name="usage"></a>
 
-## Sample Usages: Obtain Access Grant, Probe Scope, and Access Data <a name="usages"></a>
+The following pseudocode shows the SDK used with a Node JS Express server. This code walks through:
```
Comment (dtisza1) [inline readme.md:101]:
```suggestion
The following example code shows the SDK used with a Node JS Express server. This code walks through:
```

I think pseudocode should be replaced here, since these are actual coding examples.

Comment (ajshred) [reply readme.md:101]: I debated about this as well.  In one of my drafts I removed it, but it does require some setup for the code to run.  But I agree,  we can probably change this as Dave shows above.
